compose/config.py

```diff
@@ -86,7 +86,8 @@ def make_service_dict(self, name, service_dict):
         service_dict['name'] = name
         service_dict = resolve_environment(service_dict, working_dir=self.working_dir)
         service_dict = self.resolve_extends(service_dict)
-        return process_container_options(service_dict, working_dir=self.working_dir)
+        service_dict = process_container_options(service_dict, working_dir=self.working_dir)
+        return resolve_build_path(service_dict, working_dir=self.working_dir)
```

Comment:
Could we instead call `resolve_build_path` from `process_container_options`? I think that'd be conceptually cleaner.